algo/src/main/java/com/hartwig/serve/sources/ckb/blacklist/CkbBlacklistEvidenceFile.java

```diff
@@ -0,0 +1,87 @@
+
+public class CkbBlacklistEvidenceFile {
+
+    private static final String FIELD_DELIMITER = "\t";
+
+    private CkbBlacklistEvidenceFile() {
+    }
+
+    @NotNull
+    public static List<CkbBlacklistEvidenceEntry> read(@NotNull String ckbBlacklistEvidenceTsv) throws IOException {
+        List<String> lines = Files.readAllLines(new File(ckbBlacklistEvidenceTsv).toPath());
+        Map<String, Integer> fields = SerializationUtil.createFields(lines.get(0), FIELD_DELIMITER);
+
+        // Skip header
+        return fromLines(lines.subList(1, lines.size()), fields);
+    }
+
+    @NotNull
+    private static List<CkbBlacklistEvidenceEntry> fromLines(@NotNull List<String> lines, @NotNull Map<String, Integer> fields) {
+        List<CkbBlacklistEvidenceEntry> blacklistedEntries = Lists.newArrayList();
+        for (String line : lines) {
+            blacklistedEntries.add(fromLine(line, fields));
+        }
+        return blacklistedEntries;
+    }
+
+    @NotNull
+    private static CkbBlacklistEvidenceEntry fromLine(@NotNull String line, @NotNull Map<String, Integer> fields) {
+        String[] values = line.split(FIELD_DELIMITER);
+        CkbBlacklistEvidenceType type = CkbBlacklistEvidenceType.valueOf(values[fields.get("blacklistType")]);
+
+        String therapy = null;
+        String cancerType = null;
+        String gene = null;
+        String event = null;
+        EvidenceLevel level = null;
+
+        if (Sets.newHashSet(CkbBlacklistEvidenceType.EVIDENCE_BASED_ON_THERAPY,
+                CkbBlacklistEvidenceType.EVIDENCE_ON_THERAPY_AND_CANCER_TYPE,
+                CkbBlacklistEvidenceType.EVIDENCE_BASED_ON_THERAPY_AND_CANCER_TYPE_AND_GENE,
+                CkbBlacklistEvidenceType.EVIDENCE_BASED_ON_THERAPY_AND_CANCER_TYPE_AND_GENE_AND_EVENT).contains(type)) {
+            therapy = values[fields.get("therapyName")];
+        }
+        if (Sets.newHashSet(CkbBlacklistEvidenceType.EVIDENCE_ON_THERAPY_AND_CANCER_TYPE,
+                CkbBlacklistEvidenceType.EVIDENCE_BASED_ON_THERAPY_AND_CANCER_TYPE_AND_GENE,
+                CkbBlacklistEvidenceType.EVIDENCE_BASED_ON_THERAPY_AND_CANCER_TYPE_AND_GENE_AND_EVENT).contains(type)) {
+            cancerType = values[fields.get("cancerType")];
+        }
+        if (Sets.newHashSet(CkbBlacklistEvidenceType.EVIDENCE_BASED_ON_THERAPY_AND_CANCER_TYPE_AND_GENE,
+                CkbBlacklistEvidenceType.EVIDENCE_BASED_ON_THERAPY_AND_CANCER_TYPE_AND_GENE_AND_EVENT,
+                CkbBlacklistEvidenceType.ALL_EVIDENCE_BASED_ON_GENE,
+                CkbBlacklistEvidenceType.ALL_EVIDENCE_BASED_ON_GENE_AND_EVENT).contains(type)) {
+            gene = values[fields.get("gene")];
+        }
+        if (Sets.newHashSet(CkbBlacklistEvidenceType.ALL_EVIDENCE_BASED_ON_GENE_AND_EVENT,
+                CkbBlacklistEvidenceType.EVIDENCE_BASED_ON_THERAPY_AND_CANCER_TYPE_AND_GENE_AND_EVENT).contains(type)) {
+            event = values[fields.get("event")];
+        }
```

Comment:
You could consider defining these Sets as a one-off somewhere with a descriptive name.